main/src/ca/uwaterloo/flix/language/phase/EntryPoints.scala

```diff
@@ -568,15 +591,42 @@ object EntryPoints {
 
   }
 
-    /** Returns a new root where [[TypedAst.Root.entryPoints]] contains all entry points (main/test/export). */
-    def findEntryPoints(root: TypedAst.Root): TypedAst.Root = {
-      val s = mutable.Set.empty[Symbol.DefnSym]
-      for ((sym, defn) <- root.defs if isEntryPoint(defn)(root)) {
-        s += sym
-      }
-      val entryPoints = s.toSet
-      root.copy(entryPoints = Some(entryPoints))
+  /** Returns a new root where [[TypedAst.Root.entryPoints]] contains all entry points (main/test/export). */
+  def findEntryPoints(root: TypedAst.Root): TypedAst.Root = {
+    val s = mutable.Set.empty[Symbol.DefnSym]
+    for ((sym, defn) <- root.defs if isEntryPoint(defn)(root)) {
+      s += sym
+    }
+    val entryPoints = s.toSet
+    root.copy(entryPoints = Some(entryPoints))
+  }
+
+  sealed trait EvalRes[+T] {
+    def map[B](f: T => B): EvalRes[B] = this match {
+      case EvalRes.Ok(x) => EvalRes.Ok(f(x))
+      case EvalRes.ErrorOrMalformed => EvalRes.ErrorOrMalformed
+    }
+
+    def flatMap[B](f: T => EvalRes[B]): EvalRes[B] = this match {
+      case EvalRes.Ok(x) => f(x)
+      case EvalRes.ErrorOrMalformed => EvalRes.ErrorOrMalformed
     }
+  }
+
+  object EvalRes {
```

Comment:
Any reason not to use Option or Result?

Reply:
I think our own Result would fit here.

Reply:
I had separate cases for Error and Malformed previously - but it also avoids option blindness or whatever it is called. You want me to just use Result?

Reply:
In this case it would also be `Result[T, Unit]` or `Option`

Reply:
I think the nicest thing you can do is:

```
case object MyError
Result[t, MyError]
```

Its precise, monadic, and extendable.

Reply:
That's true, but sometimes you also don't like that I have results of options etc. so I don't think its black and white.

In this case I have no error information, so here it would just be
```
case object ErrorOrMalformed
Result[T, ErrorOrMalformed]
```
should I do that change?

Reply:
Seems better than what it is

Reply:
I've implemented it now - I forgot that the type of `ErrorOrMalformed` is `ErrorOrMalformed.type`